aten/src/ATen/core/ivalue.h

```diff
@@ -795,6 +818,7 @@ struct CAFFE2_API IValue final {
     int64_t as_int;
     double as_double;
     bool as_bool;
+    c10::complex<double> as_complexdouble = c10::complex<double>(0, 0);
```

Comment:
Unfortunately, this is going to increase the size of the ivalue struct, as a complex double takes up 128 bits, but all of the prior payloads in the union only took 64bits. This might be OK, but the performance implications of this will need to be scrutinized carefully. cc @ngimel @robieta

Reply:
Also, does `c10::complex<double>` not have a default constructor?

Reply:
cc. @gchanan

Reply:
I think passing around complex values directly as IValue in interpreter should be pretty rare, so I think this should be an acceptable workaround for now.

Reply:
do we have any way of evaluating this?  It's probably obvious if it matters, no?

Also, for getting the value-typeness correct you probably want a const-variant, a-la `const std::string`.

Reply:
JIT team would probably best know what "IValue overhead interpreter benchmarks" there are. But none of them are going to make use of complex numbers even today, so you'll have to exercise that case specifically.